tests/lib/prepare.sh

```diff
@@ -790,6 +790,34 @@ EOF
                            --output "$IMAGE_HOME/$IMAGE"
     rm -f ./pc-kernel_*.{snap,assert} ./pc_*.{snap,assert} ./snapd_*.{snap,assert}
 
+    if is_core20_system; then
+        # (ab)use ubuntu-seed
+        LOOP_PARTITION=2
+    else
+        LOOP_PARTITION=3
+    fi
+
+    # expand the uc16 and uc18 images a little bit (400M) as it currently will
+    # run out of space easily from local spread runs if there are extra files in
+    # the project not included in the git ignore and spread ignore, etc.
+    if ! is_core20_system; then
+        dd if=/dev/zero bs=1M count=400 >> "$IMAGE_HOME/$IMAGE"
+        # fix the GPT table because old versions of parted complain about this and
+        # refuse to properly run the next command unless the GPT table is updated
+        sgdisk "$IMAGE_HOME/$IMAGE" -e
+
+        # use parted to get the last part of free space in the image
+        # (aka the zeros we just added)
+        END_FREE_SPACE_PARTITION=$(parted -s --machine "$IMAGE_HOME/$IMAGE" print free | grep "free;$" | tail -n1)
+
+        # read the fields of the partition so we can get where the end of the free space
+        # is (which is the 3rd field in parted machine parsable output)
+        IFS=':' read -ra PARTITION <<< "$END_FREE_SPACE_PARTITION"
+
+        # resize the partition to go to the end of the free space
+        sudo parted -s "$IMAGE_HOME/$IMAGE" resizepart ${LOOP_PARTITION} "${PARTITION[2]}"
```

Comment:
Is `sudo` necessary?

Reply:
Not really, removed it now.

Reply:
OTOH, i think we could forgo the bit that computes the size and instead use `parted .. resizepart ${LOOP_PARTITION} 100%` that should resize it to take the rest of remaining space

Reply:
ah that's beautiful I honestly never even knew that parted supports percentages, this will make my local scripts where these functions came from even better!

Also all the instances of sudo were just because I had this from my local script toolbelt and as such had sudo before all the commands.